src/components/tree-view/stories/tree-view-item-exemple.tsx

```diff
@@ -73,6 +73,53 @@ export const TreeViewItemExemple = ({ ...props }: TreeViewItemExempleProps) => {
     },
   ];
 
+  const handleOpenMenu = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const handleButtonKeyDown = (e: React.KeyboardEvent<HTMLElement>) => {
+    if (e.key === "Enter" || e.key === " ") {
+      e.preventDefault();
+      setIsOpen(!isOpen);
+    }
+    if (e.key === "Escape") {
+      e.preventDefault();
+      const treeNode = e.currentTarget.closest(".c__tree-view--row");
+      if (treeNode && treeNode instanceof HTMLElement) {
+        treeNode.focus();
+      }
+    }
+    if (e.key === "ArrowRight") {
+      e.preventDefault();
+      // Navigate to next button
+      const currentButton = e.currentTarget;
+      const actionsContainer = currentButton.closest(".actions");
+      const allButtons = actionsContainer?.querySelectorAll("button");
+      if (allButtons) {
+        const currentIndex = Array.from(allButtons).indexOf(
+          currentButton as HTMLButtonElement
+        );
+        const nextIndex = (currentIndex + 1) % allButtons.length;
+        (allButtons[nextIndex] as HTMLButtonElement).focus();
+      }
+    }
+    if (e.key === "ArrowLeft") {
+      e.preventDefault();
+      // Navigate to previous button
+      const currentButton = e.currentTarget;
+      const actionsContainer = currentButton.closest(".actions");
+      const allButtons = actionsContainer?.querySelectorAll("button");
+      if (allButtons) {
+        const currentIndex = Array.from(allButtons).indexOf(
+          currentButton as HTMLButtonElement
+        );
+        const prevIndex =
+          currentIndex === 0 ? allButtons.length - 1 : currentIndex - 1;
+        (allButtons[prevIndex] as HTMLButtonElement).focus();
+      }
```

Comment:
Ah ok, it is something that we have to implement in the example, I thought in my initial comment that it could be implemented here: https://github.com/suitenumerique/ui-kit/blob/8cc52c374edc16cee1df3b86725d841030976a4c/src/components/tree-view/TreeView.tsx#L331-L339

Nit, maybe we can DRY a bit?


```suggestion
    if (e.key === "ArrowRight" || e.key === "ArrowLeft") {
      e.preventDefault();
      const currentButton = e.currentTarget;
      const actionsContainer = currentButton.closest(".actions");
      const allButtons = actionsContainer?.querySelectorAll("button");
      if (allButtons) {
      const currentIndex = Array.from(allButtons).indexOf(
        currentButton as HTMLButtonElement
      );
      const nextIndex = e.key === "ArrowRight" 
        ? (currentIndex + 1) % allButtons.length
        : currentIndex === 0 ? allButtons.length - 1 : currentIndex - 1;
      (allButtons[nextIndex] as HTMLButtonElement).focus();
      }
    }
```

Reply:
> Ah ok, it is something that we have to implement in the example, I thought in my initial comment that it could be implemented here:
> 
> https://github.com/suitenumerique/ui-kit/blob/8cc52c374edc16cee1df3b86725d841030976a4c/src/components/tree-view/TreeView.tsx#L331-L339
> 

Hm yes, you’re right, the implementation inside the example only works there. To avoid duplicating the same logic in our apps, I suggested in the fixup to extract it into a useArrowRoving hook. This way, it works well in the example and can also be reused directly in our apps.

WDYT ?